lib/horde/registry.ex

```diff
@@ -18,11 +18,18 @@ defmodule Horde.Registry do
   @doc """
   Child spec to enable easy inclusion into a supervisor:
   supervise([
-    [Horde, :node_1]
+    {Horde, id: MyId, name: MyName}
   ])
   """
-  def child_spec(node_id) do
-    %{id: node_id, start: {GenServer, :start_link, [__MODULE__, node_id]}}
+  def child_spec(options \\ []) do
+    options = Keyword.put_new(options, :id, __MODULE__)
+
+    %{
+      id: options[:id],
+      start:
+        {GenServer, :start_link,
+         [__MODULE__, Keyword.delete(options, :id), Keyword.take(options, [:name])]}
```

Comment:
What if someone forgets to supply a `name:` option? Should we guard against it?

Reply:
Good question. What does `Registry` do?

Reply:
I looked it up: `Registry` uses `name` as the `id` of the process in the `child_spec`. I'll duplicate this behaviour in `Horde.Registry`.